src/main/java/com/example/s3practical_task/controller/S3Sync.java

```diff
@@ -0,0 +1,71 @@
+package com.example.s3practical_task.controller;
+
+import com.amazonaws.services.s3.AmazonS3;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Objects;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class S3Sync {
+
+    private final AmazonS3 s3;
+
+    public void sync(String folderPath) {
```

Comment:
think on the efficiency of method from performance and cost standpoint, what we can do better?